mkversion.sh

```diff
@@ -32,6 +32,12 @@ if [ ! -z "$1" ]; then
     o=shell
 fi
 
+# If the version is passed in the environment during build, let's use that
```

Comment:
I don't think we want environment variable, positional parameters, or flag parameters that provide a version into this script. We want this script to tell the version based on the git data, and we wan this version to be statically saved into the filesystem so that it can be packed. If we have vendor tarballs that do not contain their own version today, these tarballs are broken and should be fixed. We want the source code to know its version, instead of expecting the person that builds the tarball to keep its version in mind and pass it in via a parameter. That latter approach will **definitely result in binaries reporting their version incorrectly eventually**.